packages/aws-rfdk/lib/core/scripts/powershell/configureCloudWatchAgent.ps1

```diff
@@ -16,62 +16,104 @@ param (
-  if ( $ca_thumbprint -ne '614D271D9102E30169822487FDE5DE00A352B01D' ) {
-    Write-Output "CA Thumbprint failed verification. Has the CA certificate been changed? If so, then please submit a PR to fix this."
-    Write-Output $( $ca | Format-List )
-    Exit 1
-  }
+  if (-Not $s) {
+    $gpg_keyring = "$env:temp\keyring.gpg"
 
-  # Amazon.com certificate is:
-  #  - Subject: 'CN=Amazon.com Services LLC, OU=Software Services, O=Amazon.com Services LLC, L=Seattle, S=Washington, C=US'
-  #  We do not check the Thumbprint because it will change when the certificate is rotated. This seems to be yearly.
-  $cert = $sig | Select-Object -ExpandProperty 'SignerCertificate'
-  $cert_subject = $cert | Select-Object -ExpandProperty 'Subject'
-  if ( $cert_subject -ne 'CN=Amazon.com Services LLC, OU=Software Services, O=Amazon.com Services LLC, L=Seattle, S=Washington, C=US' ) {
-    Write-Output "Amazon.com is not the issuer of this installer. If this is the correct then please submit a PR to fix this."
-    Write-Output $( $cert | Format-List )
-    Exit 1
+    # Download GPG
+    $gpg_installer = "$env:temp\gnupg-w32-2.2.27_20210111.exe"
+    wget https://gnupg.org/ftp/gcrypt/binary/gnupg-w32-2.2.27_20210111.exe -OutFile $gpg_installer
+
+    # Verify GPG
+    $gpg_sig = Get-AuthenticodeSignature $gpg_installer
+    $status = $gpg_sig | Select-Object -ExpandProperty 'Status'
+    if ( $status -ne 'Valid' ) {
+      Write-Output "GPG installer does not have a valid signature."
+      Exit 1
+    }
+    $sha256 = Get-FileHash $gpg_installer -Algorithm SHA256
+    if ( $sha256 -inotmatch '5D89E239790822711EAE2899467A764879D21440AB68E9413452FA96CEDEBA50') {
+      Write-Output "GPG failed checksum verification. Expected sha256 to be '5D89E239790822711EAE2899467A764879D21440AB68E9413452FA96CEDEBA50' but found:"
+      Write-Output $sha256
+      Exit 1
+    }
+
+    # Install GPG
+    Start-Process -Wait -FilePath $gpg_installer -ArgumentList "/S /v/qn" -PassThru
+
+    # Refresh the PATH so gpg is available
+    $env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine")
+
+    # Download Amazon's public key and import it to GPG's keyring
+    $cloudwatch_pub_key = "$env:temp\amazon-cloudwatch-agent.gpg"
+    Read-S3Object -BucketName amazoncloudwatch-agent -Key assets/amazon-cloudwatch-agent.gpg -File $cloudwatch_pub_key -Region us-east-1
+    gpg --no-default-keyring --keyring $gpg_keyring --import $cloudwatch_pub_key
+
+    # Verify that the imported key has the correct fingerprint
+    $fingerprint = '937616F3450B7D806CBD9725D58167303B789C72'
+    $keys = gpg --no-default-keyring --keyring $gpg_keyring -k
+    $keys = ($keys | Out-String).Trim() -replace "\r", "" -replace "\n", "" -replace "\[", "" -replace "\]", ""
+
+    if ($keys -inotlike "*$fingerprint*") {
+      Write-Output "Expect CloudWatch agent's public key to equal 937616F3450B7D806CBD9725D58167303B789C72 but got:"
```

Comment:
nit: Expect --> Expected.

Also, should we reference the the `$fingerprint` variable in the string instead of duplicating it? I believe Powershell supports variable expansion in strings like this:
```ps
Write-Output "Expect CloudWatch agent's public key to equal ${fingerprint} but got:"
```

Reply:
Fixed. You actually don't need the `{}` for the variable expansion, you can see it being done on the line above.